### ISSUE TITLE
C# .NET - Access Violation Exception

### ISSUE THREAD

**Issue by viniciuschiele** [zeromq/libzmq#103]
Hi,

I'm trying to use more workers in the receipt of messages, but is experiencing the following error:

`
Unhandled Exception: System.AccessViolationException: Attempted to read or write
 protected memory. This is often an indication that other memory is corrupt.
   at nzmq.libzmq.zmq_device(Int32 flag, IntPtr inSocket, IntPtr outSocket)
   at Server.Sub.Main(String[] args)
`

And sometimes this error also occurs:

`
Assertion failed: false (......\src\object.cpp:305)
`

The error occurs only after receiving some messages.
Without using the 'zmq_device' works fine.

0MQ Version: 2.0.10
Windows 7 64 bits
.NET Framework 4.0

```
static void Main(string[] args)
{
    using (NzmqContext context = new NzmqContext())
    {
        LibzmqSocket inSocket = context.CreateRawSocket(NzmqSocketTypes.Sub);
        inSocket.SetSockOptAsString(NzmqSocketOptions.Subscribe, "");
        inSocket.Connect("tcp://127.0.0.1:1000");

        LibzmqSocket outSocket = context.CreateRawSocket(NzmqSocketTypes.Push);
        outSocket.Bind("inproc://workers");

        Task[] tasks = new Task[2];

        for (int i = 0; i < tasks.Length; i++)
        {
            tasks[i] = Task.Factory.StartNew(ReceiveThread, context);
        }

        libzmq.zmq_device(libzmq.STREAMER, inSocket.Handle, outSocket.Handle);
    }

}

private static void ReceiveThread(object obj)
{
    NzmqContext context = (NzmqContext)obj;

    LibzmqSocket socket = context.CreateRawSocket(NzmqSocketTypes.Pull);
    socket.Connect("inproc://workers");

    while (true)
    {
        byte[] buffer = socket.Recv();

        Console.WriteLine(buffer.Length);
    }
}
```

Thx!
